device/cuda/include/traccc/cuda/utils/thread_id.hpp

```diff
@@ -8,34 +8,34 @@
 
 #pragma once
 
-#include <cstddef>
-
 #include "traccc/definitions/qualifiers.hpp"
 
 namespace traccc::cuda {
 struct thread_id1 {
     TRACCC_DEVICE thread_id1() {}
 
-    std::size_t inline TRACCC_DEVICE getLocalThreadId() const {
+    unsigned int inline TRACCC_DEVICE getLocalThreadId() const {
         return threadIdx.x;
     }
 
-    std::size_t inline TRACCC_DEVICE getLocalThreadIdX() const {
+    unsigned int inline TRACCC_DEVICE getLocalThreadIdX() const {
         return threadIdx.x;
     }
 
-    std::size_t inline TRACCC_DEVICE getGlobalThreadId() const {
+    unsigned int inline TRACCC_DEVICE getGlobalThreadId() const {
         return threadIdx.x + blockIdx.x * blockDim.x;
     }
 
-    std::size_t inline TRACCC_DEVICE getGlobalThreadIdX() const {
+    unsigned int inline TRACCC_DEVICE getGlobalThreadIdX() const {
```

Comment:
🤔 I think it would be fair to put in an assertion into this function that this value would not be larger than `unsigned int`. At the same time, notice that the calculation on line 30 works with `unsigned int` itself. If the result of that overflows, the `std::size_t` return type will not do anything to help. 🤔